sql/src/main/java/io/crate/metadata/doc/DocIndexMetaData.java

```diff
@@ -160,6 +166,14 @@ public static Version getVersionUpgraded(Map<String, Object> mappingMap) {
         return Version.fromMap(getNested(versionMap, Version.Property.UPGRADED.toString(), null));
     }
 
+    public static boolean isClosed(IndexMetaData indexMetaData, Map<String, Object> mappingMap, boolean isPartitioned) {
+        // First, we must check that if the table is partitioned, whether the metadata flag is there.
```

Comment:
Instead of stating what the code below does, it would be nice to comment on *why* it does it that way. (Mention the empty partitioned table case - etc.)